src/client/ui/status-bar/index.js

```diff
@@ -350,27 +355,34 @@ export default class StatusBar {
         });
     }
 
-    _showDebuggingStatus () {
+    _showDebuggingStatus (callback) {
         this.debugging = true;
 
         this.statusDiv.textContent = DEBUGGING_TEXT;
         this.buttons.style.display = 'inline-block';
 
         this._recalculateSizes();
 
-        return new Promise(resolve => {
-            eventUtils.bind(this.continueButton, browserUtils.isTouchDevice ? 'touchstart' : 'click', () => {
-                this._resetState();
-
-                resolve();
-            });
-
-            eventUtils.bind(this.nextActionButton, browserUtils.isTouchDevice ? 'touchstart' : 'click', () => {
-                this._resetState();
-
-                resolve(true);
-            });
-        });
+        var eventName = browserUtils.isTouchDevice ? 'touchstart' : 'mousedown';
+
+        this.continueButtonHandler = e => {
+            eventUtils.preventDefault(e);
+            this._resetState();
+            eventUtils.unbind(this.continueButton, eventName, this.continueButtonHandler);
+            eventUtils.unbind(this.nextActionButton, eventName, this.nextActionButtonHandler);
+            callback();
+        };
+
+        this.nextActionButtonHandler = e => {
+            eventUtils.preventDefault(e);
+            this._resetState();
+            eventUtils.unbind(this.continueButton, eventName, this.continueButtonHandler);
+            eventUtils.unbind(this.nextActionButton, eventName, this.nextActionButtonHandler);
+            callback(true);
+        };
+
+        eventUtils.bind(this.continueButton, eventName, this.continueButtonHandler);
+        eventUtils.bind(this.nextActionButton, eventName, this.nextActionButtonHandler);
```

Comment:
Hmm.. I can't understand, why have you replaced `Promise` with `callback` here?
And why do we need to add the continueButtonHandler to the field of the class? Why not a simple variables?